mantidproject/mantid
# Remove SortXAxis temporary property and fix TOSCA Workflow

Only test on windows, and on Debug mode
------------------------------------------------------
**Description of work.**
Due to an issue with some TOSCA data that was not recorded correctly, SortXAxis refused to sort the data, due to it's newer validation. The fix for this was the RemoveSpectra algorithm which can remove spectra that have invalid detectors.

**To test:**
1.`Interfaces`->`Indirect`->`Data Reduction`->`ISISEnergyTransfer` tab
2. The instrument is `TOSCA`
3. Enter run number `4970`
4. Click `Run` and wait
There should be no error.



Fixes #25281 

*This does not require release notes* because it fixes an error that only occurs in debug mode.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
